torch/csrc/jit/codegen/cuda/lower_thread_predicate.cpp

```diff
@@ -20,30 +20,42 @@ namespace {
 Bool* getPredicatePerParallelType(
     ParallelType pt,
     const ThreadPredicateMap::PredicateInfo& pred_info) {
-  kir::SimplifyingIrBuilder ir_builder(GpuLower::current()->kernel());
+  kir::SimplifyingIrBuilder simplifying_ir_builder(
+      GpuLower::current()->kernel());
 
   auto pt_dim = GpuLower::current()->parallelDimensionMap().get(pt);
 
   // If pt is not used or is proven to be one, no need to predicate.
   if (pt_dim == nullptr || pt_dim->isOneInt()) {
-    return ir_builder.trueVal();
+    return simplifying_ir_builder.trueVal();
   }
-
   // When BID needs to be predicated, that means it's an output of a grid
   // reduction and only the last block index in that dimension has the right
   // value from the grid reduce.
   if (isParallelTypeBlockDim(pt) && pred_info.limited_types.get(pt)) {
-    return ir_builder
+    return simplifying_ir_builder
         .eqExpr(
-            NamedScalar::getParallelIndex(pt),
-            ir_builder.subExpr(
-                NamedScalar::getParallelDim(pt), ir_builder.oneVal()))
+            IrBuilder::create<NamedScalar>(
+                GpuLower::current()->kernel()->as<IrContainer>(),
+                stringifyThread(pt),
+                DataType::Int),
+            simplifying_ir_builder.subExpr(
+                IrBuilder::create<NamedScalar>(
+                    GpuLower::current()->kernel()->as<IrContainer>(),
+                    stringifyThreadSize(pt),
+                    DataType::Int),
+                simplifying_ir_builder.oneVal()))
```

Comment:
The previous code looks better to me as it's more succinct, though I'm fine with the new one as well.

Reply:
This does get cleaned up in a later PR, uncertain which one.